nuxeo-core/nuxeo-core-cache/src/main/java/org/nuxeo/ecm/core/cache/CacheServiceImpl.java

```diff
@@ -241,14 +251,22 @@ public int getApplicationStartedOrder() {
 
     @Override
     public void start(ComponentContext context) {
-        // register cache invalidator
-        String nodeId = Framework.getProperty(NODE_ID_PROP);
-        if (StringUtils.isBlank(nodeId)) {
-            log.info("No cache invalidator registered");
-        } else {
+        if (Framework.isBooleanPropertyTrue(CLUSTERING_ENABLED_PROP)) {
+            // register cache invalidator
+            String nodeId = Framework.getProperty(NODE_ID_PROP);
+            if (StringUtils.isBlank(nodeId)) {
+                nodeId = String.valueOf(RANDOM.nextLong());
+                log.warn("Missing cluster node id configuration, please define it explicitly "
+                        + "(usually through repository.clustering.id). Using random cluster node id instead: "
+                        + nodeId);
+            } else {
+                nodeId = nodeId.trim();
+            }
             invalidator = new CachePubSubInvalidator();
             invalidator.initialize(CACHE_INVAL_PUBSUB_TOPIC, nodeId);
             log.info("Registered cache invalidator for node: " + nodeId);
+        } else {
+            log.info("No cache invalidator registered");
```

Comment:
Shouldn't the message rather be "Clustering is not enabled"?

Reply:
You're right I clarified the message a bit: `Not registering a cache invalidator because clustering is not enabled`